argo-cd-apps/base/member/infra-deployments/crossplane-control-plane/crossplane-control-plane.yaml

```diff
@@ -0,0 +1,41 @@
+apiVersion: argoproj.io/v1alpha1
+kind: ApplicationSet
+metadata:
+  name: crossplane-control-plane
+spec:
+  generators:
+    - merge:
+        mergeKeys:
+          - nameNormalized
+        generators:
+          - clusters:
+              values:
+                sourceRoot: components/crossplane-control-plane
+                environment: development
```

Comment:
The value here needs to be `staging`. If you run `kustomize build --enable-helm argo-cd-apps/overlays/staging-downstream` (or `konflux-public-staging`) you'll notice those overlays aren't patching this field. The development and production overlays are the ones which patch it.